applications/surgical_scene_recon/operators/endonerf_loader_op.py

```
@@ -197,6 +182,39 @@ def _load_file_lists(self, data_path):
+        num_depths = len(self.depth_paths)
+        num_masks = len(self.mask_paths)
+
+        usable_frames = min(num_poses, num_images, num_depths, num_masks)
+
+        if usable_frames == 0:
+            raise ValueError(
+                f"[EndoNeRFLoader] No usable frames: "
+                f"poses={num_poses}, images={num_images}, "
+                f"depths={num_depths}, masks={num_masks}. "
+                f"Check dataset directory: {data_path}"
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)